R/cf.R

```diff
@@ -355,6 +417,112 @@ gen.block.array <- function(n, R, l, endcorr=TRUE) {
   return(list(starts = st, lengths = lens))
 }
 
+#' Computes the samples for reweighted correlation function
+#'
+#' @param cf `cf` object.
+#' @param rw `rw` object.
+#' @param nsamples Integer (Number of stochastic samples)
+#' @param boot.R Integer
+#' @param boot.l Integer
+#' @param seed Integer
+#' @param sim string
+#' @param endcorr boolean
+#' @export
+bootstrap_rw.cf <- function(cf, rw, nsamples=1, boot.R=400, boot.l=2, seed=1234, sim="geom", endcorr=TRUE) {
+  stopifnot(inherits(cf, 'cf_orig'))
+  stopifnot(inherits(rw, 'rw_orig'))
+  stopifnot(inherits(rw, 'rw_meta'))
+  stopifnot(inherits(cf, 'cf_indexed'))
+
+
+  ##We should also check that the cf object and the rw object contains the same gauge configurations
+
+
+  stopifnot(rw$conf.index == cf$conf.index)
+ 
+  stopifnot( nrow(cf$cf) ==nsamples*length(rw$conf.index) )
+
+  ##Average over the samples
+
+  tmp <- lapply(X=1:ncol(cf$cf),FUN=function(rw_idx){
+                                       new<- matrix(cf$cf[,rw_idx],nrow=nsamples,ncol=length(rw$conf.index));
+                                       new2 <- apply(new,2,mean);
+                                       new2
+                                    }
+               )
+  cf$cf <- do.call(cbind,tmp)
+
+  boot.l <- ceiling(boot.l)
+  boot.R <- floor(boot.R)
+
+  stopifnot(boot.l >= 1)
+  stopifnot(boot.l <= nrow(cf$cf))
+  stopifnot(boot.R >= 1)
+
+  ##Construct correlation function for the reweighting samples
+  rw_cf <- cf
+  rw_cf$cf <- replicate(ncol(cf$cf), rw$rw)
+
+  ## save random number generator state
+  if (exists(".Random.seed", envir = .GlobalEnv, inherits = FALSE))
+    temp <- get(".Random.seed", envir = .GlobalEnv, inherits = FALSE)
+  else
+    temp <- NULL
+
+  ## we set the seed for reproducability and correlation
+  old_seed <- swap_seed(seed)
+
+  ## now we bootstrap the correlators
+  cf.tsboot <- boot::tsboot(cf$cf, statistic = function(x){ return(apply(x, MARGIN=2L, FUN=mean))},
```

Comment:
since you are bootstrapping here, why not require an already-bootstrapped ```cf```?

Reply:
I want to be sure, that the seed for bootstrapping is the same. Actually bootstrapping the original correlation function I do not need at all, so I will remove this